Mission file structure, refactoring, more dynamic settings.

theace0296 opened the pull request:
I couldn't even describe every file that changed, but the basic idea was to refactor and reorganize the whole mission. Everything is a function of some sort now, with the exceptions of external libraries (AIS, COS, etc) and a few legecy files that redirect to their function counterparts (AddBadPoints, etc).

Feature changes/additions (That I can think of):
- Almost all globals are declared in fn_initGlobalVars and fn_initUnitVars, replacing the #includes in a lot of files.
- Added option for setting East, Ind, and West factions, available factions and their loadouts are automatically parsed from the ConfigFile.
- Added check before anything else gets done in fn_initGlobalVars for "laptop1", if it isn't found then the mission will run a series of functions to have the admin/sl pick a position for an HQ. Then the required objects are spawned in around that position. Not everything spawned has a 1 to 1 with the manually created missions, but the functionality is the same. Only required objects in mission file are Playable Blu Units, with the Civilian Playables and the HCs optional. 
- Added check in AIS for global variable "AIS_MOD_ENABLED", which allows anyone to quickly disable AIS. Specifically I usually run a mod version of AIS, so I don't want them to overwrite each other, but I still want TRGM to be able to call AIS functions.

I'm very aware that I didn't include every file from the master branch, and would like to include as much as I can now that the hard part is done. However, if you'd rather keep your current structure, then I'd be happy to just keep this refactored version as a fork for my own use.

theace0296 committed:
refactoring

I couldn't even describe every file that changed, but the basic idea was to refactor and reorganize the whole mission. Everything is a function of some sort now, with the exceptions of external libraries (AIS, COS, etc) and a few legecy files that redirect to their function counterparts (AddBadPoints, etc).

theace0296 committed:
typos, and some extra checks for the two helos being replaced at start.

theace0296 commented:
@treendy I left this as a draft so it could be tested more before a merge of any kind is done.

theace0296 committed:
reorganized folders, 2nd column to adv options dialog

theace0296 committed:
various bugfixes, made vehicle selection more accurate, added monsoon weather option, more functional code

theace0296 committed:
bug fix, added large patrols mission setting

theace0296 committed:
change bis_fnc_call to just call for remote exec

theace0296 committed:
log all the things

theace0296 committed:
Update initplayerlocal.sqf

theace0296 committed:
bug fixes, more logging

theace0296 committed:
wrapper for find safe pos to make it more reliable, various bug fixes

theace0296 committed:
added blizzard option, seperated time and weather mission start up param

theace0296 committed:
bug fixes and optimizations

theace0296 committed:
bug fixes and updates

treendy commented on CustomMission/TRGMSetDefaultMissionSetupVars.sqf:
```diff
@@ -1,115 +0,0 @@
-
```
This file is needed, see here: https://youtu.be/K6u34Ekxsmo?t=115
(have set the video to start at 1:55, as that is where i go into detail about the values)

theace0296 replied:
Slowly working it out, but see [here](https://github.com/treendy/TRGM2.Malden/pull/134/commits/d1084649ff41c0f8abaa7bcb34ad21ff150dc9d0?file-filters%5B%5D=.sqf#diff-88745d6c427a1bc1b88927173576e07aaaeafd47f84d50b2cb4d04e788663426).

treendy commented on CustomMission/TRGMSetEnemyFaction.sqf:
```diff
@@ -1,65 +0,0 @@
-
```
This file is needed, see here: https://youtu.be/K6u34Ekxsmo?t=115
(have set the video to start at 1:55, as that is where i go into detail about the values)

treendy commented on CustomMission/TRGMSetFriendlyLoadouts.sqf:
```diff
@@ -1,329 +0,0 @@
-
```
This file is needed, see here: https://youtu.be/K6u34Ekxsmo?t=115
(have set the video to start at 1:55, as that is where i go into detail about the values)

treendy reviewed (commented):
Hey, thanks for taking the time to go through all this :)
I will take a look and comment etc... (been meaning to move everything into functions)
Thank you :)

theace0296 committed:
bug fixes and reorganizing

cleaned up redundant code in initplayerlocal, moved camera effects so all players get them.

theace0296 committed:
updates, bug fixes, incorporated forced mission set-up

theace0296 committed:
documented the gui functions

theace0296 committed:
vehicle customization for the vehicle requests menu!

theace0296 committed:
added timer option for enemy reinforcements, refactored all missions to newer format (WIP)

theace0296 committed:
bug fixes from refactoring, added debug parameter to be able to reset settings to defaults

theace0296 committed:
bug fixes, use different way of random choice to make it more clear what the odds are

theace0296 committed:
meeting assassination mission renamed, added trigger to alert units when players are at obj

theace0296 committed:
bug fixes

theace0296 committed:
bug fixes, added ambush convoy mission

convoy mission is mostly working, I'd like to see the initial spawning not randomly blow up vehicles but ¯\_(ツ)_/¯, also the AI pathing is 😂

theace0296 committed:
chill out some of the waitUntil's, convoy mission bug fixes

theace0296 committed:
added download/hacking progress bar, various other mission tweaks and bug fixes

theace0296 committed:
refactoring of initplayerlocal, bug fixes

theace0296 committed:
notification system, bug fixes, refactoring

theace0296 committed:
bug fixes and move initplayerlocal to function (easier to debug)

theace0296 committed:
Moved global vars to initGlobalVars

theace0296 committed:
removed block macros from zen occupy house to please linter

theace0296 committed:
Create universal task complete function, DRY programing

theace0296 committed:
OP Treb. Unit exceptions, debug mode will create 1 of each objective

theace0296 committed:
missing semi colon

theace0296 committed:
mission bug fixes, general bug fixes

theace0296 committed:
all factions available for all sides, new mission parameter storage

theace0296 committed:
change == to isEqualTo, alternate update task methods, bug fixes

theace0296 committed:
Bug fixes and support for new DLC

theace0296 committed:
Add mission file templates and build tools

theace0296 committed:
Publish workflow

theace0296 committed:
Update env var

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Testing build

theace0296 committed:
Only pbo files for publish

theace0296 committed:
Use publish folder for artifact

theace0296 committed:
lint action

theace0296 committed:
lint action

theace0296 committed:
lint action

theace0296 committed:
lint action use pwsh

theace0296 committed:
unix duh, just use rm

theace0296 committed:
unix syntax

theace0296 committed:
sqflint args

theace0296 committed:
sqflint exclusions

theace0296 committed:
disable lint for now

theace0296 committed:
Create README.md

theace0296 committed:
Rename

theace0296 committed:
Use logo

theace0296 committed:
Custom factions (hopefully)

theace0296 committed:
Bug fixes

theace0296 committed:
Rename prepackaged missions

theace0296 committed:
update templates

theace0296 committed:
Update README.md

theace0296 committed:
General bug fixes and improvements

theace0296 committed:
Merge branch 'refactor' of https://github.com/theace0296/TRGM2.Malden into refactor

theace0296 committed:
Update README.md

theace0296 committed:
Bug fixes

theace0296 committed:
Remove addons deps

theace0296 committed:
Restructure by Client/Server/Global/Var

theace0296 committed:
Fix errors and addon deps

theace0296 committed:
Try enabling linter again

theace0296 committed:
No gitignore anymore

theace0296 committed:
Rename publish artifact

theace0296 committed:
Lint task still doesn't work

theace0296 committed:
bug fixes

theace0296 committed:
replace tabs with spaces

theace0296 committed:
fix lines

theace0296 committed:
Validation pipeline

theace0296 committed:
paths

theace0296 committed:
no BOM check

theace0296 committed:
args for lint

theace0296 committed:
Allow selecting unit/veh type in requests dialog

theace0296 committed:
Workspace updates

theace0296 committed:
Remove extraneous brace

theace0296 committed:
Create nightly.yml

theace0296 committed:
Use nighlty release

theace0296 commented:
Since my version has evolved into something so different in function than the original TRGM2, it would be extremely hard to just "merge" the two together. For now, I'm breaking my fork into it's own repo, let me know if you would want to attempt to merge the two together at some point.
